plugins/block-plus-minus/src/procedures.js

```diff
@@ -388,65 +389,75 @@ const procedureDefMutator = {
    */
   validator_: function(newName) {
     const sourceBlock = this.getSourceBlock();
-    // Replaces all whitespace (including non-breaking) with normal spaces.
-    // Then removes spaces at the beginning or end of the string.
-    newName = newName.replace(/[\s\xa0]+/g, ' ').replace(/^ | $/g, '');
-    if (!newName) {
-      return null;
-    }
+    const workspace = sourceBlock.workspace;
+    const argData = sourceBlock.argData_;
+    const argDatum = sourceBlock.argData_.find(
+        (element) => element.argId == this.name);
+    const currId = argDatum.model.getId();
+
+    // Replace all whitespace with normal spaces, then trim.
+    newName = newName.replace(/[\s\xa0]+/g, ' ').trim();
+    const caselessName = newName.toLowerCase();
 
     /**
-     * Checks that all of the args (that aren't this arg) have a different
-     * name than this arg.
-     * @param {{model: Blockly.VariableModel, argId:string}} element
-     * @return {boolean} True if the other name is not a match.
+     * Returns true if the given argDatum is the argDatum associated with this
```

Comment:
Would it still be correct if you changed this to:
"Returns true if the given argDatum is associated with this field, or has a different caseless name than the argDatum associated with this field."